### PR TITLE
Allow setting default role for new joiners

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Introduces an org setting for specifiying the default role of new joiners.

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/ENT-39/allow-collaborator-as-default-role-for-new-joiners

## How to test

Join the org https://se-collaborator.preview.gitpod-dev.com/orgs/join?inviteId=94727dca-2121-4cbc-a8f0-c2bab788993b

You should be an owner

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/service/json-rpc-organization-client.ts [hunk 1]
```diff
@@ -249,7 +251,10 @@ export class JsonRpcOrganizationClient implements PromiseClient<typeof Organizat
                 "updateRestrictedEditorNames is required to be true to update restrictedEditorNames",
             );
         }
-        await getGitpodService().server.updateOrgSettings(request.organizationId, update);
+        await getGitpodService().server.updateOrgSettings(request.organizationId, {
+            ...update,
+            defaultRole: request.defaultRole as OrgMemberRole,
```
Comment (geropl) [inline components/dashboard/src/service/json-rpc-organization-client.ts:256]: This line seems superfluous :monocle_face:

**Comment by geropl** [comment]
Starting to review now :eyes:

File: components/dashboard/src/service/json-rpc-organization-client.ts (hunk 1 above)
Comment (svenefftinge) [reply components/dashboard/src/service/json-rpc-organization-client.ts:256]: it casts string to OrgMemberRole

Comment (geropl) [reply components/dashboard/src/service/json-rpc-organization-client.ts:256]: Ah, different types, yummy :grimacing:

**Comment by geropl** [comment]
Seeing this when trying to store org settings in my own org: 
image

**Comment by geropl** [comment]
After using the dropDown once it works.

File: components/server/src/orgs/organization-service.ts [hunk 2]
```diff
@@ -310,8 +312,12 @@ export class OrganizationService {
                     const isDataOps = await getExperimentsClientForBackend().getValueAsync("dataops", false, {
                         teamId: orgId,
                     });
+                    console.log("isDataOps", isDataOps);
+                    console.log("orgSettings.defaultRole", orgSettings.defaultRole);
```
Comment (geropl) [inline components/server/src/orgs/organization-service.ts:316]: Debug output?

File: components/gitpod-db/src/typeorm/entity/db-team-settings.ts [hunk 3]
```diff
@@ -30,6 +30,9 @@ export class DBOrgSettings implements OrganizationSettings {
     @Column("json", { nullable: true })
     restrictedEditorNames?: string[] | null;
 
+    @Column("varchar", { nullable: true })
```
Comment (geropl) [inline components/gitpod-db/src/typeorm/entity/db-team-settings.ts:33]: I'd really really love us to not use null, as I never saw it working as expected. :confused: 
E.g. this is what we get back from the DB layer when [storing an empty shape](https://github.com/gitpod-io/gitpod/tree/gpl/se/collaborator):
```
      + expected - actual

      -{
      -  "defaultRole": null,
      -  "defaultWorkspaceImage": null,
      -  "allowedWorkspaceClasses": null,
      -  "pinnedEditorVersions": null,
      -  "restrictedEditorNames": null,
      -  "workspaceSharingDisabled": false
      -}
      +{}
  ```

Given that we use the same pattern already I'd say it's outside the scope of this PR, though... :cry:

**Review by geropl — APPROVED** [review]
Tested on the provided and my own orgs, and works as expected :heavy_check_mark: :sparkles: 

I'd be awesome if we could align on how to do DB changes, but don't want to hijack/block this PR. Will post in Slack next week.

**Pull request merged**